scripts/ansible/deploy_jenkins.yml

```diff
@@ -13,6 +13,25 @@
      - include_tasks: tasks/ansible-install-sgx-packages.yml
      - include_tasks: tasks/ansible-install-az-dcap-client.yml
 
+     - name: Provision new linux slave node
+       jenkins_script:
+         script: "{{ lookup('template', '../templates/linux_slave.groovy.j2') }}"
+         user: "{{ jenkins_master_user }}"
+         password: "{{ jenkins_master_password }}"
+         url: "{{ jenkins_master }}"
+
+     - name: Extract slave secret
+       jenkins_script:
+         script: jenkins.model.Jenkins.getInstance().getComputer("{{ jenkins_slave_name }}").getJnlpMac()
+         user: "{{ jenkins_master_user }}"
+         password: "{{ jenkins_master_password }}"
+         url: "{{ jenkins_master }}"
+       register: script_output
+
+     - set_fact: node_secret="{{script_output.output | replace('Result'+':','') | trim}}"
```

Comment:
This looks a little bit ugly.

Instead of:
```
script_output.output | replace('Result'+':','') | trim
```
I'd suggest that we use:
```
script_output.output.lstrip('Result:') | trim
```
Looks a lot cleaner. From what I see so far, groovy scripts outputs have "Result: <blabla>" pattern. The [lstrip](https://python-reference.readthedocs.io/en/latest/docs/str/lstrip.html) function removes the leading characters and it does the work very well here.

Reply:
Replace with the suggestion.